usecases/schema/tenant.go

```diff
@@ -188,15 +189,23 @@ func (m *Manager) DeleteTenants(ctx context.Context, principal *models.Principal
 		return fmt.Errorf("multi-tenancy is not enabled for class %q", class)
 	}
 
-	// create transaction payload
-	tenantNames := make([]string, len(tenants))
-	for i, tenant := range tenants {
-		tenantNames[i] = tenant.Name
+	m.shardingStateLock.Lock()
```

Comment:
I agree that checking for duplicate tenants as part of input validation makes sense. However, it seems more reasonable not to verify the existence of the actual physical shard during deletion. The delete operation should be idempotent, allowing multiple processes to attempt the deletion of the same tenants without issues. Moreover, this approach can be beneficial in repairing nodes that may fail during the deletion process

Reply:
good point, changed